File: c-droid/repl.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include "utils.c"
#include "engine.c"
#include "table.h"

void parse_command(Table* table, char *command) {
    trim_spaces(command);
    if (strcmp(command, ".exit") == 0) {
        db_close(table);
        printf("exiting.. good bye!\n");
        exit(0);
    } else if (strcmp(command, ".help") == 0) {
        printf("Available commands:\n");
        printf(".exit - Exit the program\n");
        printf(".help - Show this help message\n");
    } else {
        execute_sql(table, command);
    }
}

File: zig-droid/repl.zig
const std = @import("std");

pub fn parse_command(command: []const u8) void {
    const trm_cmd = std.mem.trim(u8, command, " \n");
    if (std.mem.eql(u8, trm_cmd, ".exit")) {
        std.debug.print("Exiting REPL...\n", .{});
        std.os.exit(0);
    } else if (std.mem.eql(u8, trm_cmd, ".help")) {
        std.debug.print("Available commands:\n", .{});
        std.debug.print(".exit - Exit the REPL\n", .{});
        std.debug.print(".help - Show this help message\n", .{});
    } else {
        std.debug.print("Unknown command: {s}\n", .{trm_cmd});
    }
}
